### DISCUSSION TITLE
Database installation permissions

### DISCUSSION THREAD

**Discussion by camilogarciabotero** [oschwengers/bakta#83] [Q&A] [answered]
Hi @oschwengers,

Thanks again for working on Bakta. I was updating the Bakta database to v.3.0 in a cluster and came across that I did not have user permissions to make the update smoothly. I used the command `bakta_db update -d BAKTA_DB` and it worked nice, but since I did not have permissions on the installed database, then the update crashes. As I did not change the permission any time before, I think that Bakta is downloading and decompressing them with root permissions.  

¿So I was wondering if there is any way you can help sort this issue, to avoid back and forth communication with facilities asking for permissions. Or it was just my case?

Thanks in advance,
Camilo.

**Comment by oschwengers** [comment, marked as answer]
Hi @camilogarciabotero ,
of course, that always depends on who downloaded the database first and whether that happened as user or root. In addition, using the db `update` mode, Bakta requires write permissions either for the current working directory or the tmp dir that might have been provided via `--tmp-dir`. Could you provide the entire CLI output of the update process?

As `v3.0` is a major update, you could also just remove the old database anyways and use `download` to get a fresh db instance.

**Reply by camilogarciabotero** [reply]
Hi @oschwengers,

Thanks for your quick reply. I downloaded the database as a user of the cluster and I can see that during the process the `db.tar.gz` has permissions for the users. However, once it enters the decompression step only root permission is granted for the files inside de `db` folder. I've used both the `bakta_db download --output <output-path>` as well as the manual option:

```
wget https://zenodo.org/record/5215743/files/db.tar.gz
tar -xzf db.tar.gz
rm db.tar.gz
amrfinder_update --force_update --database db/amrfinderplus-db/
```

Both scenarios end up with the database folder `db` files having only root access, so my question is if there is a way to end up with the `db` files with the user permissions, can the `tar` step be settled to decompress files with the user's permission? or if this is a matter of how the files were compressed before being downloaded, can this be changed?

This is because any time I will want to update the `db`, then I will have to ask the cluster service to grant me access.

Thanks again and hope I've been more clear.

**Reply by oschwengers** [reply]
I tested the download on a both local and remote machine manually and via `bakta download`. Unfortunately, I cannot reproduce this behavior. So without further information like logs and output / tmp directory permissions I currently have no idea what might go wrong here. Is the download dir owned by root? Are there maybe some sticky bits set on this output directory?
If this is the case, could you also try that within a directory that is not owned by root but yourself? Could you maybe also ask your admins to execute the `bakta download` command (as root) and set the right permissions afterwards?

**Reply by camilogarciabotero** [reply]
Jum well, this is weird since I had the db v2.0 in the same folder that I created with user permissions so this is owned. The actual log is fine since the download ends fine:

```
drwxr-xr-x 3 4.0K Nov  2 13:31 db
```

```
cd db
```

```
drwxr-xr-x 3   50 Nov  2 12:49 amrfinderplus-db
-r--r--r-- 1 3.5M Aug  4 15:11 antifam.h3f
-r--r--r-- 1 9.4M Aug  5 02:26 expert-protein-sequences.dmnd
-r--r--r-- 1  74K Aug  5 01:53 ncRNA-genes.i1i
-r--r--r-- 1  13M Aug  5 02:01 ncRNA-regions.i1m
-r--r--r-- 1 4.5M Aug  5 02:01 ncRNA-regions.i1p
-r--r--r-- 1 116M Aug  5 08:33 pfam.h3f
-r--r--r-- 1 500K Aug  5 08:33 pfam.h3i
-r--r--r-- 1 220M Aug  5 08:33 pfam.h3p
-r--r--r-- 1  19G Nov  2 12:48 psc.dmnd
-r--r--r-- 1  89K Aug  5 01:53 rfam-go.tsv
-r--r--r-- 1 177K Aug  4 15:05 rRNA.i1f
-r--r--r-- 1  309 Aug  4 15:05 rRNA.i1i
-r--r--r-- 1  721 Aug  9 07:16 version.json
```

Now I see this is kind of problematic as bakta needs to read those files to actually run... I just tried and got this (which is completely expected as for the file permissions):

```
ERROR: database file (antifam.h3i) not readable!
ERROR: database file (antifam.h3i) not readable!
ERROR: database file (antifam.h3i) not readable!
ERROR: database file (antifam.h3i) not readable!
ERROR: database file (antifam.h3i) not readable!
```

I will try asking the admins again if they could run `bakta_db download` and see the behavior...

Thank you!
